app/controllers/flood_risk_engine/enrollments_controller.rb

```diff
@@ -2,11 +2,49 @@
 
 module FloodRiskEngine
   class EnrollmentsController < ApplicationController
+
     def new
-      enrollment = Enrollment.create
-      # TODO: make first step come from state_machine
-      url = enrollment_step_path(enrollment, enrollment.initial_step)
-      redirect_to url
+      @enrollment = Enrollment.new
+
+      @form = form_object
     end
+
+    def create
+      @enrollment = Enrollment.new
+
+      @form = form_object
+
+      if(@form.validate(params))
+        # This is essentially the start of the journey, send new Enrollment in init state
+        # to the Step Controller to take over
+        if(enrollment_params["location_check"] == "yes")
+
+          @enrollment = Enrollment.create
+          @enrollment.go_forward
+          @enrollment.save
+
+          redirect_to  enrollment_step_path(@enrollment, @enrollment.step)
+        else
+          # This effectively ends the journey
+          redirect_to main_app.page_path('contact_ea_location')
+        end
+      else
+        # Perhaps should happen in Reform Form validation - we must have an answer
+        render :new
+      end
+
+    end
+
+    private
+
+    # The form object factory dont work for new Enrollment sso have to roll our own
+    def form_object
+      "FloodRiskEngine::Steps::#{@enrollment.current_step.classify}Form".constantize.factory(@enrollment)
+    end
```

Comment:
As this isn't really a step might it be worth moving the form object from `forms/flood_risk_engine/steps` to `forms/flood_risk_engine` and then just accessing it as FloodRiskEngine::CheckLocationForm?